File: evidence/generated-systems/angular/analytics-on-angular/src/app/models/SemanticModel.js

// Define collection and schema for SemanticModel
export interface SemanticModel {
    name:
	type : string
    version:
	type : string
    grain:
	type : string
    Datasets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataSet' }]
    Metrics:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Metric' }]
    Dimensions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Dimension' }]
    Measures:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Measure' }]
    GlossaryTerms:
 	type : [{ type: Schema.Types.ObjectId, ref: 'BusinessGlossaryTerm' }]
#
    collection: 'semanticModels'
}

File: evidence/generated-systems/rubyonrails/analytics-on-rails/analyticsonrails/app/models/semanticmodel.rb
class SemanticModel < ApplicationRecord


  has_many :Datasets, class_name: 'DataSet'
  has_many :Metrics, class_name: 'Metric'
  has_many :Dimensions, class_name: 'Dimension'
  has_many :Measures, class_name: 'Measure'
  has_many :GlossaryTerms, class_name: 'BusinessGlossaryTerm'

end
